File: ios/MazadatImagePicker.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MazadatImagePicker, NSObject)

RCT_EXTERN_METHOD(openCamera:(int)length lang:(NSString *)lang
                 resolve:(RCTPromiseResolveBlock)resolve
                 reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(editPhoto:(NSString *)path lang:(NSString *)lang
                 resolve:(RCTPromiseResolveBlock)resolve
                 reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(openIdVerification:(NSString *)lang
                 resolve:(RCTPromiseResolveBlock)resolve
                 reject:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/MazadatImagePicker.swift
import UIKit
@objc(MazadatImagePicker)
class MazadatImagePicker: NSObject {

    @objc
    func openCamera(_ length:Int, lang: String, resolve : @escaping  RCTPromiseResolveBlock,reject : @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async{
            let controller = CameraController()
            UIView.appearance().semanticContentAttribute = lang != "ar" ? .forceLeftToRight : .forceRightToLeft
            controller.setData(length: length, lang: lang)
            controller.setPromise(promise: resolve)
            controller.modalPresentationStyle = .fullScreen
            UIApplication.shared.windows.filter {$0.isKeyWindow}.first?.rootViewController?.present(controller, animated: true, completion: nil)
        }
      
    }
    
    @objc
    func editPhoto(_ path:String,selectedIndex: Int, lang: String, resolve : @escaping  RCTPromiseResolveBlock,reject : @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async{
            let controller = CameraController()
            UIView.appearance().semanticContentAttribute = lang != "ar" ? .forceLeftToRight : .forceRightToLeft
            controller.setData(path: path.components(separatedBy: ","),selectedIndex: selectedIndex, lang: lang)
            controller.setPromise(promise: resolve)
            controller.modalPresentationStyle = .fullScreen
            UIApplication.shared.windows.filter {$0.isKeyWindow}.first?.rootViewController?.present(controller, animated: true, completion: nil)
        }
      
    }
    
    @objc
    func openIdVerification(_ lang: String, resolve : @escaping  RCTPromiseResolveBlock,reject : @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async{
            let controller = CameraController()
            UIView.appearance().semanticContentAttribute = lang != "ar" ? .forceLeftToRight : .forceRightToLeft
            controller.setData(length: 2, lang: lang)
            controller.setPromise(promise: resolve)
            controller.setIsIdVerification(isIdVerification: true)
            controller.modalPresentationStyle = .fullScreen
            UIApplication.shared.windows.filter {$0.isKeyWindow}.first?.rootViewController?.present(controller, animated: true, completion: nil)
        }
      
    }
}
